### ISSUE TITLE
Cannot Save component even if the fields are unique.

### ISSUE THREAD

**Issue by vaidikMirae** [strapi/strapi#17604]
### Required System information

- Node.js version: 18.12
- NPM version: 9.6
- Strapi version: v4.12.1
- Database: Postgresql
- Operating system: Linux
- Is your project Javascript or Typescript: Javascript

### Describe the bug

When I try to save the single content type, which has a links component with a unique field that throws an error anytime I try to save the content, it throws an error even if I provide the unique content.
### Steps to reproduce the behavior

1. Go to single type content
2.  Create a component with a unique field inside it.
3.  Fill the content in single type
4.  Try to save the content.

### Expected behavior

Save the single type content with the unique fields inside it.

### Screenshots

![image](https://github.com/strapi/strapi/assets/140403673/a6076370-c885-4eb3-9036-929922eb2c72)

**Comment by pjr94** [comment]
I have also experienced this issue. I have a ParticipantID field which relates to an ID provided by an external system, so I cannot simply change the ID in order to save the Entry as it would be desynced with the external system. I've not run into the issue yet, but if the client asks me to update some data in the collection (eg phone number or whatever), I will not be able to do this without changing the ParticipantID, which would be a big problem and break the relationship with the external system unless they can update it there too. Either way, its a bad work around and unproffessional.

TL:DR Yes it is very important to fix this for business reasons!

The Issue:
I think the issue is more than the unique field is comparing its new content with the existing old content and recognising it as not unique. 

Solution: There needs to be an exemption that unique fields can be the same in the same when it relates to the same Entry, so you can edit the Entry without changing the unique field.

**Comment by jhoward1994** [comment]
This issue has been fixed in v5 from 5.0.0-beta.13 onwards via https://github.com/strapi/strapi/pull/20427
